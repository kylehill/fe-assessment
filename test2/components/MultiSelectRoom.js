const Dropdown = ({
    title,
    value,
    options,
    enabled,
    handleChange
}) => {
    return (
        <>
            <div className="room-dropdown">
                <div className="room-dropdown-title">
                    {title}
                </div>
                <select disabled={!(enabled)} onChange={handleChange} 
                    value={value}>
                    {options.map((opt) => {
                        return <option key={opt} value={opt}>{opt}</option>
                    })}
                </select>
            </div>
            <style jsx>{`
                .room-dropdown {
                    display: inline-block;
                    width: 50px;
                    margin: 10px 5px;
                }

                .room-dropdown select {
                    margin-top: 5px;
                }
            `}</style>
        </>
    )
}

const MultiSelectRoom = ({ 
    number, 
    room, 
    changeOccupants, 
    toggleRoom
}) => {    
    return (
        <>
            <li className={`room ${room.enabled ? "" : "room-disabled"}`}>
                <header>
                    {number > 0 && (
                        <input type="checkbox" 
                            checked={room.enabled}
                            onChange={(e) => {
                                toggleRoom(number, e.target.checked)
                            }} />
                    )}
                    Room {number + 1}<br/>
                </header>
                <section>
                    <Dropdown title="Adults (18+)" 
                        value={room.adults}
                        options={[1, 2]}
                        enabled={room.enabled}
                        handleChange={(e) => {
                            changeOccupants(number, e.target.value, room.children)
                        }} />
                    <Dropdown title="Children (0-17)" 
                        value={room.children}
                        options={[0, 1, 2]}
                        enabled={room.enabled}
                        handleChange={(e) => {
                            changeOccupants(number, room.adults, e.target.value)
                        }} />
                </section>
            </li>
            <style jsx>{`
                .room {
                    width: 125px;
                    display: inline-block;
                    border: solid 2px #E7E7E7;
                    margin-right: 20px;
                    border-radius: 5px;
                }

                .room.room-disabled {
                    border: solid 2px #CBCFDA;
                    background: #DBDBE3;
                }

                .room header {
                    background: #E7E7E7;
                    padding: 5px;
                    font-weight: bold;
                }

                .room.room-disabled header {
                    background: #DBDBE3;
                }

                .room header input {
                    margin-right: 5px;
                }
            `}</style>
        </>
    )
}

MultiSelectRoom.defaultProps = {
    number: 0,
    room: {},
    changeOccupants: () => {},
    toggleRoom: () => {}
}

export default MultiSelectRoom