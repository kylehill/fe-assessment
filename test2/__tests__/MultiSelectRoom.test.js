import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import MultiSelectRoom from '../components/MultiSelectRoom'

describe("Smoketest", () => {
  it("renders the MultiSelectRoom component", () => {
    const component = shallow(
        <MultiSelectRoom 
            number={0}
            room={{ enabled: true, adults: 1, children: 0 }} />
    )
  })
})

describe("Snapshot", () => {
  it("matches the existing snapshot", () => {
    const component = renderer.create(<MultiSelectRoom number={0} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})