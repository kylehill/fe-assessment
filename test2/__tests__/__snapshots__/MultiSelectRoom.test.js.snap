// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches the existing snapshot 1`] = `
<li
  className="jsx-1384089475 room room-disabled"
>
  <header
    className="jsx-1384089475"
  >
    Room 
    1
    <br
      className="jsx-1384089475"
    />
  </header>
  <section
    className="jsx-1384089475"
  >
    <div
      className="jsx-4138645028 room-dropdown"
    >
      <div
        className="jsx-4138645028 room-dropdown-title"
      >
        Adults (18+)
      </div>
      <select
        className="jsx-4138645028"
        disabled={true}
        onChange={[Function]}
      >
        <option
          className="jsx-4138645028"
          value={1}
        >
          1
        </option>
        <option
          className="jsx-4138645028"
          value={2}
        >
          2
        </option>
      </select>
    </div>
    <div
      className="jsx-4138645028 room-dropdown"
    >
      <div
        className="jsx-4138645028 room-dropdown-title"
      >
        Children (0-17)
      </div>
      <select
        className="jsx-4138645028"
        disabled={true}
        onChange={[Function]}
      >
        <option
          className="jsx-4138645028"
          value={0}
        >
          0
        </option>
        <option
          className="jsx-4138645028"
          value={1}
        >
          1
        </option>
        <option
          className="jsx-4138645028"
          value={2}
        >
          2
        </option>
      </select>
    </div>
  </section>
</li>
`;
